### PR TITLE
Add Apache API tests

### PR DESCRIPTION
**Pull request by disegmvm**
Changes include:
1. Apache API tests
2. Testng file update

Run `mvn clean test -Ptestng` to run added Apache API tests with TestNG runner.

### CODE REVIEW

File: src/test/java/com/epam/rp/api/tests/ApacheApiTest.java [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+    public void testCreateDashboard_ValidParameters() throws IOException {
+        HttpClient client = HttpClients.createDefault();
+        HttpPost postRequest = new HttpPost(baseUrl + "/api/v1/" + projectName + "/dashboard");
+        postRequest.setHeader("Authorization", "Bearer " + apiToken);
+        postRequest.setHeader("Content-type", "application/json");
+        String randomDescription = "Desc " + UUID.randomUUID();
+        String randomName = "Name " + UUID.randomUUID();
+        String jsonBody = String.format("{ \"description\": \"%s\", \"name\": \"%s\" }", randomDescription, randomName);
+        postRequest.setEntity(new StringEntity(jsonBody));
+        HttpResponse response = client.execute(postRequest);
```
Comment (VerkhovtsovPavel) [inline src/test/java/com/epam/rp/api/tests/ApacheApiTest.java]: Looks good, I would suggest to introduce a set of utility methods for avoid code duplication:
e.g:
```java
private HttpResponse post(String url, String jsonBody) {
        HttpClient client = HttpClients.createDefault();
        HttpPost postRequest = new HttpPost(url);
        postRequest.setHeader("Authorization", "Bearer " + apiToken);
        postRequest.setHeader("Content-type", "application/json");
        postRequest.setEntity(new StringEntity(jsonBody));
        return client.execute(postRequest);
}
```

and same for all other used types of requests. I believe auth headers setting also could be unified, but the initial update should bring significant code reduction.

File: src/test/java/com/epam/rp/api/tests/ApacheApiTest.java [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+        String jsonBody = String.format("{ \"description\": \"Test description\", \"name\": \"\" }");
+        postRequest.setEntity(new StringEntity(jsonBody));
+        HttpResponse response = client.execute(postRequest);
+        int statusCode = response.getStatusLine().getStatusCode();
+
+        Assert.assertEquals(statusCode, 400);
+        String responseBody = EntityUtils.toString(response.getEntity());
+        ObjectMapper objectMapper = new ObjectMapper();
+        JsonNode rootNode = objectMapper.readTree(responseBody);
+        String errMessage = rootNode.path("message").asText();
```
Comment (VerkhovtsovPavel) [inline src/test/java/com/epam/rp/api/tests/ApacheApiTest.java]: This lines can be moved to separate method and reused across the tests.

```java
private String getBodyNode(HttpsResponse response, String path) {
        String responseBody = EntityUtils.toString(response.getEntity());
        ObjectMapper objectMapper = new ObjectMapper();
        JsonNode rootNode = objectMapper.readTree(responseBody);
        return rootNode.path(path).asText();
}
```

**Review by VerkhovtsovPavel — APPROVED** [review]

**Comment by VerkhovtsovPavel** [comment]
General note. The tests is highly dependent on the API engine, it's much more suitable for code maintainability to create interfaces for send base types of request and validate responses (status code, body). As a next step create implementation/s for required tool/s. It helps decouple tests from engine and simplify code updates.

File: src/test/java/com/epam/rp/api/tests/ApacheApiTest.java (hunk 2 above)
Comment (disegmvm) [reply src/test/java/com/epam/rp/api/tests/ApacheApiTest.java]: Hi Pavel, it's updated, thank you

File: src/test/java/com/epam/rp/api/tests/ApacheApiTest.java (hunk 1 above)
Comment (disegmvm) [reply src/test/java/com/epam/rp/api/tests/ApacheApiTest.java]: Thanks for your comment, it is updated now

**Pull request merged**